include/zephyr/mgmt/ec_host_cmd/backend.h

```diff
@@ -21,98 +21,110 @@
-	 *  command framework and handler (when handler_owns). Buffer is owned
-	 *  by devices and lives as long as device is valid. Device will never
-	 *  read from this buffer (for security reasons).
+	/**
+	 * Buffer to hold received data. The buffer is provided by the handler if
+	 * EC_HOST_CMD_HANDLER_RX_BUFFER > 0. Otherwise, the backend should provide
+	 * the buffer on its own and overwrites @a buf pointer in the init function.
 	 */
 	uint8_t *buf;
-	/** Number of bytes written to @a buf by device (when dev_owns). */
-	size_t *len;
-	/** Device will take when it needs to write to @a buf and @a size. */
-	struct k_sem *dev_owns;
-	/** Handler will take so it can read @a buf and @a size */
-	struct k_sem *handler_owns;
+	/** Number of bytes written to @a buf by backend. */
+	size_t len;
+	/**
+	 * The backend gives @a handler_owns, when data in @a buf are ready.
+	 * The handler takes @a handler_owns to read data in @a buf.
+	 */
+	struct k_sem handler_owns;
 };
 
 /**
- * @brief Context for host command backend and framework to pass tx data
+ * @brief Context for host command backend and handler to pass tx data
  */
 struct ec_host_cmd_tx_buf {
-	/** Data to write to the host */
+	/**
+	 * Data to write to the host The buffer is provided by the handler if
+	 * EC_HOST_CMD_HANDLER_TX_BUFFER > 0. Otherwise, the backend should provide
+	 * the buffer on its own and overwrites @a buf pointer and @a len_max
+	 * in the init function.
+	 */
 	void *buf;
-	/** Number of bytes to write from @a buf */
+	/** Number of bytes to write from @a buf. */
 	size_t len;
+	/** Size of @a buf. */
+	size_t len_max;
 };
 
-typedef int (*ec_host_cmd_backend_api_init)(
-	const struct device *dev, struct ec_host_cmd_rx_ctx *rx_ctx);
-
-typedef int (*ec_host_cmd_backend_api_send)(
-	const struct device *dev,
-	const struct ec_host_cmd_tx_buf *tx_buf);
-
-__subsystem struct ec_host_cmd_backend_api {
-	ec_host_cmd_backend_api_init init;
-	ec_host_cmd_backend_api_send send;
+/**
+ * This structure is common to all backends and is expected to be
+ * the first element in a backend-specific config structure.
+ */
+struct ec_host_cmd_config {
+	/** Backend device instance. */
+	const struct device *dev;
 };
 
 /**
- * @brief Initialize a host command device
+ * @brief Initialize a host command backend
  *
- * This routine initializes a host command device, prior to its first use. The
- * receive context object are an output of this function and are valid
- * for the lifetime of this device. The RX context is used by the client to
- * receive data from the host.
+ * This routine initializes a host command backend. It includes initialization
+ * a device used to communication and setting up buffers.
+ * This function is called by the ec_host_cmd_init function. However, if
+ * DT_CHOSEN(zephyr_host_cmd_backend) is defined, the ec_host_cmd_init function
+ * should be called automatically by a backend using SYS_INIT with
+ * CONFIG_EC_HOST_CMD_INIT_PRIORITY priority.
```

Comment:
No need to rehash the ec_host_cmd_init() requirements again.
```suggestion
 * This function is called by the ec_host_cmd_init function.
```